velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -431,5 +431,43 @@ TEST_F(DateTimeFunctionsTest, addMonths) {
       addMonths("2023-07-10", kMax),
       fmt::format("Integer overflow in add_months(2023-07-10, {})", kMax));
 }
+
+TEST_F(DateTimeFunctionsTest, nextDay) {
+  const auto nextDayFunc = [&](const std::optional<Date> date,
+                               const std::optional<StringView> dayOfWeek) {
+    return evaluateOnce<Date>("next_day(c0, c1)", date, dayOfWeek);
+  };
+
+  const auto nextDay = [&](const std::string& dateStr, const auto& dayOfWeek) {
+    Date d0;
+    parseTo(dateStr, d0);
+    return nextDayFunc(d0, dayOfWeek);
+  };
+
+  const auto parseDateStr = [&](const std::string& dateStr) {
+    Date d0;
+    parseTo(dateStr, d0);
+    return d0;
+  };
+
+  EXPECT_EQ(nextDay("2015-07-23", "Mon"), parseDateStr("2015-07-27"));
+  EXPECT_EQ(nextDay("2015-07-23", "mo"), parseDateStr("2015-07-27"));
+  EXPECT_EQ(nextDay("2015-07-23", "Tue"), parseDateStr("2015-07-28"));
+  EXPECT_EQ(nextDay("2015-07-23", "tu"), parseDateStr("2015-07-28"));
+  EXPECT_EQ(nextDay("2015-07-23", "we"), parseDateStr("2015-07-29"));
+  EXPECT_EQ(nextDay("2015-07-23", "wed"), parseDateStr("2015-07-29"));
+  EXPECT_EQ(nextDay("2015-07-23", "Thu"), parseDateStr("2015-07-30"));
+  EXPECT_EQ(nextDay("2015-07-23", "TH"), parseDateStr("2015-07-30"));
+  EXPECT_EQ(nextDay("2015-07-23", "Fri"), parseDateStr("2015-07-24"));
+  EXPECT_EQ(nextDay("2015-07-23", "fr"), parseDateStr("2015-07-24"));
```

Comment:
It may be better to cover cross month/year cases, e.g., nextDay("2015-12-31", "Mon").